data/core/commands/doc.lua

```diff
@@ -36,9 +37,23 @@ local function save(filename)
     filename = core.normalize_to_project_dir(filename)
     abs_filename = core.project_absolute_path(filename)
   end
-  doc():save(filename, abs_filename)
-  local saved_filename = doc().filename
-  core.log("Saved \"%s\"", saved_filename)
+  local ok = pcall(doc().save, doc(), filename, abs_filename)
+  if ok then
+    local saved_filename = doc().filename
+    core.log("Saved \"%s\"", saved_filename)
+  else
+    core.nag_view:show("Saving failed", string.format("Could not save \"%s\" do you want to save to another location?", doc().filename), {
+      { font = style.font, text = "No", default_no = true },
+      { font = style.font, text = "Yes" , default_yes = true }
+    }, function(item)
+      if item.text == "Yes" then
+        core.root_view:defer_draw(function()
+          -- this seems hella dumb but it wouldnt work without the defer
+          command.perform("doc:save-as")
```

Comment:
I wonder if we should also set the `active_view` to the `DocView` before calling `doc:save-as`, to avoid race issues with change in focus.

Reply:
> It's because NagView does some funky stuff with focus (force_focus) that makes the CommandView unable to be shown.
We should improve how NagView works, but this is something to think about after we implement floating Views.

ahh ok, ill add the proper comment yeah